lib/WP_Auth0_LoginManager.php

```diff
@@ -587,6 +576,68 @@ public function end_session() {
 		}
 	}
 
+	/**
+	 * Get and filter the scope used for access and ID tokens
+	 *
+	 * @param string $context - how are the scopes being used?
+	 *
+	 * @return string
+	 */
+	public static function get_userinfo_scope( $context = '' ) {
+		return apply_filters( 'auth0_auth_token_scope', 'openid email name nickname picture', $context );
+	}
+
+	/**
+	 * Get a base authorize URL for handling HLP redirects.
+	 *
+	 * @param null|string $connection - specific connection to use, pass null to use all enabled.
+	 * @param null|string $redirect_to - URL to redirect upon successful authentication.
+	 *
+	 * @return array
+	 */
+	public static function get_authorize_params( $connection = null, $redirect_to = null ) {
+		$params = array();
+		$options = WP_Auth0_Options::Instance();
+		$lock_options = new WP_Auth0_Lock10_Options();
+		$is_implicit = (bool) $options->get( 'auth0_implicit_workflow', FALSE );
+		$nonce = WP_Auth0_Nonce_Handler::getInstance()->get();
+
+		$params[ 'client_id' ] = $options->get( 'client_id' );
+		$params[ 'scope' ] = self::get_userinfo_scope( 'authorize_url' );
+		$params[ 'response_type' ] = $is_implicit ? 'id_token': 'code';
+		$params[ 'redirect_uri' ] = $is_implicit
+			? $lock_options->get_implicit_callback_url()
+			: $options->get_wp_auth0_url( null, $is_implicit );
+
+		if ( $is_implicit ) {
+			$params[ 'nonce' ] = $nonce;
+		}
+
+		if ( ! empty( $connection ) ) {
+			$params[ 'connection' ] = $connection;
+		}
+
+		// Get the telemetry header.
+		$telemetry = WP_Auth0_Api_Client::get_info_headers();
```

Comment:
this object generates headers with lots of information, right? why use only Auth0-Client? what about the rest?

Reply:
If you look at what `WP_Auth0_Api_Client::get_info_headers()` returns, you'll see. That's meant to built telemetry headers for API calls but the value of the array it returns can be used here. That was part of my comment about it being a good candidate for refactoring but beyond the scope of this release.